packages/generator/src/generator-utils.ts

```diff
@@ -185,23 +189,37 @@ export function getGenericParameters(
   entityClassName: string,
   prop: VdmProperty
 ): string {
-  const param = prop.isCollection
-    ? prop.isComplex
-      ? [`${prop.jsType}`]
-      : [`'${prop.edmType}'`]
-    : [];
+  let param;
+  if (prop.isCollection) {
+    if (prop.isComplex) {
+      param = [`${prop.jsType}`];
+    } else if (prop.isEnum) {
+      param = ["'Edm.Enum'"];
+    } else {
+      param = [`'${prop.edmType}'`];
+    }
+  } else {
+    param = [];
+  }
   return [entityClassName, ...param].join(', ');
 }
 
 export function createPropertyFieldInitializer(
   property: VdmProperty,
   entityClassName: string
 ) {
-  const edmOrComplexType = property.isComplex
-    ? property.jsType
-    : `'${property.edmType}'`;
-  const collectionTypeOrEdmType =
-    property.isComplex && !property.isCollection ? undefined : edmOrComplexType;
+  const edmOrComplexTypeOrEnumType =
```

Comment:
This block is a bit messy I would say. I would refactor it in the following way:
- create a helper function for creating the ``new ${property.fieldType}(${[``
- This helper function needs the `isCollection` and `type:string` as input.
- handle the special case of (isCollection=true and enum) 
- Then you call this helper for the concrete three cases:
```
if(isEnum){
return helper(isCollection,property.jsType)
}
if(isComplex){
return helper(isCollection,property.jsType)
}
if(isEdm){
helper(isCollection,property. edmType)
}

Reply:
I personal feel this is hard to understand as well, but it's not easy to refactor like what you said.
Because we want to decide `collectionTypeOrEdmType` by checking 3 boolean variables (isEnum, isComplex and isCollection), which has 2^3 = 8 cases. For 3 cases out of 8, we have "special treatment" and the rest of them (5 cases), we return something like a "default value".
In this way, the if-else condition block is a bit longer than your example.
I refactored a bit, so you really see 3 "special cases" and the "default behaviour".